applications/GeoMechanicsApplication/custom_utilities/generic_utilities.h

```diff
@@ -48,6 +49,36 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GenericUtilities
         return result;
     }
 
+    template <class ContainerType, class IdsType>
+    static void CollectIdsFromEntity(const ContainerType& rContainer, IdsType& rIds)
+    {
+        for (const auto& r_item : rContainer) {
+            const auto id = r_item.Id();
+
+            if constexpr (requires { rIds.insert(id); }) {
+                // Works for set/unordered_set
+                rIds.insert(id);
+            } else if constexpr (requires { rIds.push_back(id); }) {
+                // Works for vector/deque
+                rIds.push_back(id);
+            } else {
+                static_assert(!std::is_same_v<IdsType, IdsType>,
+                              "OutputContainerType does not support insert or push_back.");
+            }
+        }
+    }
```

Comment:
I had the feeling this could be simplified, didn't manage -> asked @avdg81 and he had the nice idea of using an output iterator instead of reference to the container. Then the function would look like this:

```suggestion
    template <class ContainerType, class OutputIt>
    static void CollectIdsFromEntity(const ContainerType& rContainer, OutputIt Out)
    {
        std::transform(std::begin(rContainer), std::end(rContainer), Out,
                               [](const auto& rItem) { return rItem.Id(); });
    }
```
On the calling side that would mean we either pass a std::back_inserter for a vector or a std::inserter for a set (I'll place some comments in the relevant places to clarify)

Reply:
that is really short; however, then each call depends on the type. Hope we can discuss this tomorrow.